File: string/leetcode_520.java
package string;

public class leetcode_520 {
    public static void main(String[] args) {
        leetcode_520 lc = new leetcode_520();
        System.out.println(lc.detectCapitalUse("Leetcode"));
    }

    public boolean detectCapitalUse(String word) {
        if (word.length() < 2) return true;

        char[] chars = word.toCharArray();
        boolean[] boolArr = new boolean[3];
        boolArr[0] = chars[0] - 'a' > 0;

        for (int i = 1; i < chars.length; i++) {
            char ch = chars[i];
            int ac = ch - 'a';

            if (!boolArr[1])
                boolArr[1] = ac < 0;
            if (!boolArr[2])
                boolArr[2] = ac >= 0;

            if (boolArr[0] && ac < 0)
                return false;
            if (boolArr[1] && boolArr[2])
                return false;
        }

        return true;
    }
}

File: string/LeetCode_520.kt
package string

class LeetCode_520 {

    fun detectCapitalUse(word: String): Boolean {
        if (word.length == 1) {
            return true
        }

        if (word.first().isUpperCase()) {
            val isCapital = word[1].isUpperCase()

            for (i in 2 until word.length) {
                if (word[i].isUpperCase() != isCapital) {
                    return false
                }
            }
        } else {
            for (i in 1 until word.length) {
                if (word[i].isUpperCase()) {
                    return false
                }
            }
        }

        return true
    }
}

fun main() {
    // Input
    val word = "USA"
    // Output
    LeetCode_520().detectCapitalUse(word).let {
        println(it)
        require(it)
    }
}
